pkg/jobs/job_info_storage.go

```diff
@@ -82,7 +82,7 @@ func (i InfoStorage) get(ctx context.Context, infoKey []byte) ([]byte, bool, err
 	row, err := i.txn.QueryRowEx(
 		ctx, "job-info-get", i.txn.KV(),
 		sessiondata.NodeUserSessionDataOverride,
-		"SELECT value FROM system.job_info WHERE job_id = $1 AND info_key = $2",
+		"SELECT value FROM system.job_info WHERE job_id = $1 AND info_key = $2 LIMIT 1",
```

Comment:
May i recommend "order by written desc limit 1" because otherwise you're relying on implicit behaviors in the query optimizer

Reply:
done!